src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -201,6 +210,21 @@ internal override string GetValueToDisplay()
             {
                 return (_value == null) ? "null" : string.Format("\"{0}\"", _value);
             }
+
+            public override string ToString(string? format, IFormatProvider? provider)
+            {
+                int formatLength = RopeValue.Length;
+                if (format is not null && int.TryParse(format, out var len) &&
+                    len >= 0)
```

Comment:
>len >= 0 [](http://example.com/codeflow?start=20&length=8)

Minor: It looks like we could check `len >= 3` here, then skip the `formatLength >= 3` below.